src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,251 @@
+
+        ______TS("success: single field - startTime");
+        assertEquals(START_TIME, notificationsDb.getNotification(n.getNotificationId()).getStartTime());
+        assertEquals(START_TIME_DIFF, notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withStartTime(START_TIME_DIFF)
+                        .build()).getStartTime());
+        assertEquals(START_TIME_DIFF, notificationsDb.getNotification(n.getNotificationId()).getStartTime());
+
+        ______TS("success: single field - endTime");
+        assertEquals(END_TIME, notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+        assertEquals(END_TIME_DIFF, notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withEndTime(END_TIME_DIFF)
+                        .build()).getEndTime());
+        assertEquals(END_TIME_DIFF, notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+
+        ______TS("success: single field - shown");
+        assertFalse(notificationsDb.getNotification(n.getNotificationId()).isShown());
+        assertTrue(notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withShown()
+                        .build()).isShown());
+        assertTrue(notificationsDb.getNotification(n.getNotificationId()).isShown());
+
+        ______TS("success: bulk update");
+        // We will try to revert back to the original attribute to test bulk update
+        // Note: Update of shown is not revertable so not to check it here
+        n = notificationsDb.getNotification(n.getNotificationId());
+        assertEquals(TITLE_DIFF, n.getTitle());
+        assertEquals(MESSAGE_DIFF, n.getMessage());
+        assertEquals(TYPE_DIFF, n.getType());
+        assertEquals(TARGET_USER_DIFF, n.getTargetUser());
+        assertEquals(START_TIME_DIFF, n.getStartTime());
+        assertEquals(END_TIME_DIFF, n.getEndTime());
+        assertTrue(n.isShown());
+        notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTitle(TITLE)
+                        .withMessage(MESSAGE)
+                        .withType(TYPE)
+                        .withTargetUser(TARGET_USER)
+                        .withStartTime(START_TIME)
+                        .withEndTime(END_TIME)
+                        .build());
+
+        n = notificationsDb.getNotification(n.getNotificationId());
+        assertEquals(TITLE, n.getTitle());
+        assertEquals(MESSAGE, n.getMessage());
+        assertEquals(TYPE, n.getType());
+        assertEquals(TARGET_USER, n.getTargetUser());
+        assertEquals(START_TIME, n.getStartTime());
+        assertEquals(END_TIME, n.getEndTime());
+        assertTrue(n.isShown());
```

Comment:
Add in failure cases?

Reply:
Update method will not fail at the Db level. For the invalid parameters, checks are done at the Attributes class so shouldn't that be under AttributeTest, not DbTest?

Reply:
Defer to discussion below